src/FakeItEasy/Core/DefaultFakeManagerAccessor.cs

```diff
@@ -22,13 +23,22 @@ public FakeManager GetFakeManager(object proxy)
             Guard.AgainstNull(proxy);
 
             FakeManager? result = this.TryGetFakeManager(proxy);
+            if (result is not null)
+            {
+                return result;
+            }
 
-            if (result is null)
+            if (proxy is Delegate { Target: not null } d)

```

Comment:
Sometimes we're passed the Delegate, and sometimes the DynamicProxy proxy. If the delegate...

Actually, I think I've double-fixed this issue. Here and in the generator. I'll see if I can simplify.